src/actions/CZMStrainAction.C

```diff
@@ -0,0 +1,160 @@
+#include "AddVariableAction.h"
+#include "CZMStrainAction.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "RankTwoScalarTools.h"
+#include "libmesh/string_to_enum.h"
+registerMooseAction("DeerApp", CZMStrainAction, "add_postprocessor");
+registerMooseAction("DeerApp", CZMStrainAction, "add_material");
+registerMooseAction("DeerApp", CZMStrainAction, "meta_action");
+
+InputParameters CZMStrainAction::validParams() {
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Compute the total, normal and sliding strain "
+                             "contirbutions of a cohesive model to an RVE.");
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary",
+      "The list of boundary names where the cohesive model is present");
+  params.addParam<std::vector<SubdomainName>>(
+      "block", "The list of subdomain names representing the bulk material");
+  params.addParam<PostprocessorName>(
+      "bulk_volume_PP", "",
+      "The name of the PP representing the bulk volume at time 0.");
+  params.addParam<PostprocessorName>(
+      "czm_strain_base_name", "czm_strain",
+      "A string containing the base name for interface strain");
+  params.addParam<bool>(
+      "compute_cumulative_strain", true,
+      "If true (default) also add all integrated interface strains");
+  params.addParam<bool>("compute_equivalent_strain", true,
+                        "If true (default) also add all the equivalnet strain");
+  params.addRequiredParam<std::vector<VariableName>>(
+      "displacements", "The displacement variables");
+  params.addParam<bool>("large_kinematics", true,
```

Comment:
I default the solid mechanics kernel to false.  Does this gracefully degenerate to the small deformation behavior?  If not should default this to false as well.